Parity/src/QwHelicity.cc

```diff
@@ -1121,8 +1121,12 @@ void QwHelicity::SetEventPatternPhase(Int_t event, Int_t pattern, Int_t phase)
 void QwHelicity::SetFirstBits(UInt_t nbits, UInt_t seed)
 {
   // This gives the predictor a quick start
-  UShort_t firstbits[nbits];
-  for (unsigned int i = 0; i < nbits; i++) firstbits[i] = (seed >> i) & 0x1;
+  // At present, this routine can only handle nbits=24 (see GetRandomSeed)
+  if (nbits != 24)
+	throw std::invalid_argument("SetFirstBits currently only supports 24 bits.");
+  // Allocate nbits+1 elements as GetRandomSeed expects Fortran indexing (1-nbits)
+  UShort_t firstbits[nbits+1];  // NB firstbits[0] is never used
+  for (unsigned int i = 0; i < nbits; i++) firstbits[i+1] = (seed >> i) & 0x1;
```

Comment:
Changing the loop limit to be "nbits+1" and the indexing of firstbits to be just "i" results in passing (nearly) the same array into the GetRandomSeed function as it was before.  The "nearly" in that statement depends on the value picked up by the buffer-overflow in the old code, so depending on compiler we may still see a difference in the mockdata generated before and after this fix.

Reply:
Sure, that'll work, too. Depends on what you're trying to accomplish and whether reproducing old mock data matters. With your suggested change, the lowest bit of `seed` will not be used, and bit 24 will always be zero. Less entropy, more compatibility, I guess.

Perhaps we could add a 30-bit version, since 30 bits seems to be what the rest of the code expects?